geopandas/tests/test_geodataframe.py

```diff
@@ -393,6 +393,24 @@ def test_to_file_schema(self):
 
         assert result_schema == schema
 
+    def test_to_file_crs(self):
+        """
+        Ensure that the file is written according to the crs
+        if it is specified
+
+        """
+        tempfilename = os.path.join(self.tempdir, 'crs.shp')
+        self.df.to_file(tempfilename, crs=self.df.crs)
+        # Read layer back in
+        df = GeoDataFrame.from_file(tempfilename)
+        assert 'geometry' in df
+        assert len(df) == 5
+        assert np.alltrue(df['BoroName'].values == self.boros)
+        # add oherwise failed
+        df.crs["x_0"] = int(df.crs["x_0"])
+        df.crs['wktext'] = True
+        assert self.df.crs == df.crs
```

Comment:
I think a slightly better test of this would be to remove the `crs` completely from the dataframe before calling `to_file` (but, of course, passing in the original `crs` using the argument you're adding). As is, it looks to me like this test would pass regardless of whether the `crs` passed in by the user was actually used.

Reply:
The code without the PR changes raise an error and therefore the test fails.
But I kind of agree, I will remove the two lines in the next days.